src/Driver/PDO/Connection.php

```diff
@@ -87,17 +91,80 @@ public function quote(string $value): string
         return $this->connection->quote($value);
     }
 
-    public function lastInsertId(?string $name = null): string
+    /**
+     * {@inheritDoc}
+     */
+    public function lastInsertId()
     {
+        $driverName = $this->connection->getAttribute(PDO::ATTR_DRIVER_NAME);
+
+        if ($driverName === 'oci') {
```

Comment:
Should this be a new connection class?

Reply:
Yes, this was left as a demonstration for an alternative to subclassing. See https://github.com/doctrine/dbal/pull/4324/files#r499686504

Reply:
I'm good with this approach as well. Let's start simple, by having all the logic in one class, and introduce sub-classes if necessary.